### PR TITLE
[IMP] *: add knowledge tour to redirect to knowledge article

### PR DESCRIPTION
**Pull request by jaeschwa**
Added the knowledge tour to all industry modules so when a DB is started a tour pointing to the knowledge app is directly launched.

To avoid the maintenance lines of code for these tours, these files are added in the cloc_exclude of the manifest, leading to the automatic creation of ir_model_data entries to exclude the cloc. This change is related to this PR: odoo/odoo@a47de680de14

In all the modules, the message related to the onboarding is improved.

task-4056239

### CODE REVIEW

File: agriculture_shop/data/mail_message.xml
Comment (Jean-boje) [inline agriculture_shop/data/mail_message.xml:1]: <field name="subject">🚀 Get started with Odoo Agriculture Shop</field>

File: agriculture_shop/data/konwledge_tour.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="knowledge_tour" model="web_tour.tour">
+        <field name="name">knowledge_tour</field>
+        <field name="sequence">2</field>
+        <field name="rainbow_man_message">Congrats, enjoy your reading! </field>
```
Comment (Jean-boje) [inline agriculture_shop/data/konwledge_tour.xml:6]: For each rainbow_man_message in this PR, replace by:
<field name="rainbow_man_message">Welcome ! Happy exploring. </field>

File: art_craft/data/mail_message.xml [hunk 2]
```diff
@@ -6,11 +6,10 @@
         <field name="message_type">email</field>
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo Art Craft</field>
```
Comment (Jean-boje) [inline art_craft/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo Art & Craft Store</field>

File: architects/data/mail_message.xml [hunk 3]
```diff
@@ -6,11 +6,10 @@
         <field name="message_type">email</field>
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo Architects</field>
```
Comment (Jean-boje) [inline architects/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo Architecture Firm </field>

File: agriculture_shop/static/src/js/my_tour.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
+
+registry.category("web_tour.tours").add("knowledge_tour", {
+    url: "/odoo",
+    
+    steps: () => [
+        {
+            trigger: '.o_app[data-menu-xmlid="knowledge.knowledge_menu_root"]',
+            content: _t("Open Knowledge to find the onboarding guide."),
```
Comment (Jean-boje) [inline agriculture_shop/static/src/js/my_tour.js:12]: For each module, update content of tour:
content: _t("Get on track and explore our recommendations for your Odoo usage here!"),

File: automobile/data/mail_message.xml [hunk 5]
```diff
@@ -6,11 +6,10 @@
         <field name="message_type">email</field>
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo Automobile</field>
```
Comment (Jean-boje) [inline automobile/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo Automobile Spare Parts</field>

File: bar_industry/data/mail_message.xml [hunk 6]
```diff
@@ -7,9 +7,9 @@
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
         <field name="subject">🚀 Get started with Odoo Bar</field>
```
Comment (Jean-boje) [inline bar_industry/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo Bar & Pub</field>

File: beverage_distributor/__manifest__.py
Comment (Jean-boje) [inline beverage_distributor/__manifest__.py:1]: No mail_message.xml in Beverages ?

File: beverage_distributor/data/konwledge_tour.xml
Comment (Jean-boje) [inline beverage_distributor/data/konwledge_tour.xml:1]: Change konwledge_tour.xml into knowledge_tour.xml

File: certification_organism/data/mail_message.xml [hunk 7]
```diff
@@ -6,11 +6,10 @@
         <field name="message_type">email</field>
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo Certification Organism</field>
```
Comment (Jean-boje) [inline certification_organism/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo Audit & Certification</field>

File: clothing_boutique/data/mail_message.xml [hunk 8]
```diff
@@ -6,11 +6,10 @@
         <field name="message_type">email</field>
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo Clothing Boutique</field>
```
Comment (Jean-boje) [inline clothing_boutique/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo Clothing Store</field>

File: coal_petroleum/data/mail_message.xml [hunk 9]
```diff
@@ -6,11 +6,10 @@
         <field name="message_type">email</field>
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo Coal and Petroleum</field>
```
Comment (Jean-boje) [inline coal_petroleum/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo Fossil Fuel Trading</field>

File: condominium/data/mail_message.xml [hunk 10]
```diff
@@ -6,11 +6,10 @@
         <field name="message_type">email</field>
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo Condominium</field>
```
Comment (Jean-boje) [inline condominium/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo Property Owner Association</field>

File: hardware_shop/data/mail_message.xml [hunk 11]
```diff
@@ -6,11 +6,10 @@
         <field name="message_type">email</field>
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo Hardware Shop</field>
```
Comment (Jean-boje) [inline hardware_shop/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo Hardware Store</field>

File: headhunter/data/mail_message.xml [hunk 12]
```diff
@@ -6,11 +6,10 @@
         <field name="message_type">email</field>
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo Headhunter</field>
```
Comment (Jean-boje) [inline headhunter/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo Talent Acquisition</field>

File: industry_lawyer/data/mail_message.xml [hunk 13]
```diff
@@ -6,11 +6,10 @@
         <field name="message_type">email</field>
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo Lawyer</field>
```
Comment (Jean-boje) [inline industry_lawyer/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo Law Firm</field>

File: it_hardware/data/mail_message.xml [hunk 14]
```diff
@@ -6,11 +6,11 @@
         <field name="message_type">email</field>
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo IT Hardware</field>
```
Comment (Jean-boje) [inline it_hardware/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo IT Hardware & Support</field>

File: micro_brewery/data/mail_message.xml [hunk 15]
```diff
@@ -6,11 +6,10 @@
         <field name="message_type">email</field>
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo Micro Brewery</field>
```
Comment (Jean-boje) [inline micro_brewery/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo Microbrewery</field>

File: non_profit_organization/data/mail_message.xml [hunk 16]
```diff
@@ -6,11 +6,10 @@
         <field name="message_type">email</field>
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo Non Profit Organization</field>
```
Comment (Jean-boje) [inline non_profit_organization/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo Nonprofit Organization</field>

File: pharmacy_retail/data/mail_message.xml [hunk 17]
```diff
@@ -6,11 +6,10 @@
         <field name="message_type">email</field>
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo Pharmacy retail</field>
```
Comment (Jean-boje) [inline pharmacy_retail/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo Pharmacy</field>

File: solar_installation/data/mail_message.xml [hunk 18]
```diff
@@ -6,11 +6,10 @@
         <field name="message_type">email</field>
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo Solar Installation</field>
```
Comment (Jean-boje) [inline solar_installation/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo Energy System</field>

File: surveyor/data/mail_message.xml [hunk 19]
```diff
@@ -6,11 +6,10 @@
         <field name="message_type">email</field>
         <field name="author_id" ref="base.partner_root"/>
         <field name="subtype_id" ref="mail.mt_comment"/>
-        <field name="subject">Knowledge article</field>
+        <field name="subject">🚀 Get started with Odoo Surveyor</field>
```
Comment (Jean-boje) [inline surveyor/data/mail_message.xml:9]: <field name="subject">🚀 Get started with Odoo Surveying & Mapping</field>

File: agriculture_shop/__manifest__.py [hunk 20]
```diff
@@ -37,6 +37,7 @@
         'data/website_view.xml',
         'data/website_theme_apply.xml',
         'data/ir_model_data.xml',
+        'data/konwledge_tour.xml',
```
Comment (vava-odoo) [inline agriculture_shop/__manifest__.py:40]: typo to correct in all modules
```suggestion
        'data/knowledge_tour.xml',
```

**Review by vava-odoo — COMMENTED** [review]
Hi @jaeschwa 
Can you take the review comments into account? I also added minor things to update.
Thanks for this huge work!

File: agriculture_shop/data/konwledge_tour.xml (hunk 1 above)
Comment (vava-odoo) [reply agriculture_shop/data/konwledge_tour.xml]: I'm pretty sure there is no space here: "Welcome!"

File: custom_furniture/__manifest__.py [hunk 21]
```diff
@@ -66,6 +66,16 @@
         'demo/quality_check.xml',
         'demo/planning_slot.xml',
     ],
+    'assets': {
+        'web.assets_backend': [
+            'custom_furniture/static/src/js/my_tour.js',
+        ]
+    },
+    'author': 'Odoo S.A.',
```
Comment (vava-odoo) [inline custom_furniture/__manifest__.py:74]:
```suggestion
```

File: electronic_store/data/knowledge_article.xml [hunk 22]
```diff
@@ -163,7 +154,20 @@
                 </ul>
                 <blockquote>Use Ship Later option before the checkout for shipping later</blockquote>
                 <p><br /></p>
-            ]]>
+        ]]>
+    </template>
+
+    <record id="welcome_article" model="knowledge.article">
+        <field name="name">ELECTRONICS STORE</field>
```
Comment (vava-odoo) [inline electronic_store/data/knowledge_article.xml:161]: :eyes:

File: odoo_partner/__manifest__.py [hunk 23]
```diff
@@ -48,6 +48,16 @@
         'demo/timesheet.xml',
         'demo/project_project.xml',
     ],
+    'assets': {
+        'web.assets_backend': [
+            'odoo_partner/static/src/js/my_tour.js',
+        ]
+    },
+    'author': 'Odoo S.A.',
```
Comment (vava-odoo) [inline odoo_partner/__manifest__.py:56]:
```suggestion
```

**Review by vava-odoo — COMMENTED** [review]
Almost there :upside_down_face: 
Could you also check that mail messages are correct in beverage distributor, custom furniture and odoo partner, as they are not in the current diff?

**Review by vava-odoo — APPROVED** [review]

**Pull request merged**